backend/src/api.py

```diff
@@ -330,34 +331,36 @@ async def store_random_review(feedback_input: Request) -> OkResponse:
     return OkResponse()
 
 
-def _make_tpm_tar_file_from_results() -> Iterator[Path]:
-    if not FEEDBACK_DIR:
-        raise HTTPException(status_code=HTTPStatus.NOT_FOUND, detail="No data found")
-
-    with get_temporary_dir() as tmp_dir:
-        tar_path = make_tar(
-            f"results-{int(datetime.now().timestamp())}.tar.gz", Path(FEEDBACK_DIR), tmp_dir
-        )
-        try:
-            yield tar_path
-        finally:
-            os.unlink(tar_path)
+@app.get("/download")
+def download_results():
+    """
+    Download all results we have as a tar.gz archive.
 
+    We create a temporary directory and store the archive in there.
+    After the client browser gets the whole file, we delete our temp
+    file and directory using the cleanup() method as a backgroun task.
 
-@app.get("/download", response_class=StreamingResponse)
-def download_results(_tar_path=Depends(_make_tpm_tar_file_from_results)):
-    def iter_large_file(file_name: Path):
-        with open(file_name, mode="rb") as file:
-            yield from file
+    This function was rewritten from an async implementataion that stopped working
+    (probably after an update to fastapi and starlette).
+    https://github.com/fedora-copr/log-detective-website/issues/157
+    """
+    if not FEEDBACK_DIR:
+        raise HTTPException(status_code=HTTPStatus.NOT_FOUND, detail="No data found")
 
-    return StreamingResponse(
-        iter_large_file(_tar_path),
+    tmp_dir = Path(tempfile.mkdtemp())
+    tar_name = f"results-{int(datetime.now().timestamp())}.tar.gz"
+    tar_path = make_tar(tar_name, Path(FEEDBACK_DIR), tmp_dir)
+
+    def cleanup():
+        os.unlink(tar_path)
+        os.rmdir(tmp_dir)
+    # https://fastapi.tiangolo.com/advanced/custom-response/?h=fileresponse#fileresponse
+    # https://fastapi.tiangolo.com/reference/background/?h=background
+    return FileResponse(
```

Comment:
cool, I didn't know this wrapper around StreamingResponse exists. So much better!